botocore/httpsession.py

```diff
@@ -341,17 +342,19 @@ def _setup_ssl_cert(self, conn, url, verify):
             conn.cert_reqs = 'CERT_NONE'
             conn.ca_certs = None
 
-    def _setup_proxy_ssl_context(self, proxies_settings):
+    def _setup_proxy_ssl_context(self, proxies_settings, proxy_url):
         proxy_ca_bundle = proxies_settings.get('proxy_ca_bundle')
         proxy_cert = proxies_settings.get('proxy_client_cert')
         if proxy_ca_bundle is None and proxy_cert is None:
             return None
 
         context = self._get_ssl_context()
         try:
+            u = parse_url(proxy_url)
```

Comment:
We likely need to be accounting for `LocationParseError` if we're going to add this.

Reply:
Thanks. I will add